microprofile/reactive-messaging-smallrye/extension/src/main/java/org/wildfly/extension/microprofile/reactive/messaging/MicroProfileReactiveMessagingSubsystemDefinition.java

```diff
@@ -104,13 +112,40 @@ private AddHandler() {
         protected void performBoottime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
             super.performBoottime(context, operation, model);
 
+            installKafkaElytronSSLContextRegistryServiceIfPresent(context);
+
             context.addStep(new AbstractDeploymentChainStep() {
                 public void execute(DeploymentProcessorTarget processorTarget) {
-                    processorTarget.addDeploymentProcessor(SUBSYSTEM_NAME, DEPENDENCIES, DEPENDENCIES_MICROPROFILE_REACTIVE_MESSAGING, new ReactiveMessagingDependencyProcessor());
+                    processorTarget.addDeploymentProcessor(SUBSYSTEM_NAME, DEPENDENCIES,
+                            DEPENDENCIES_MICROPROFILE_REACTIVE_MESSAGING, new ReactiveMessagingDependencyProcessor());
+
+                    MicroProfileReactiveMessagingLogger.LOGGER.debug("Looking for DynamicDeploymentProcessorAdder implementations");
+                    Module module =
+                            ((ModuleClassLoader)WildFlySecurityManager.getClassLoaderPrivileged(this.getClass())).getModule();
+                    ServiceLoader<DynamicDeploymentProcessorAdder> sl = module.loadService(DynamicDeploymentProcessorAdder.class);
+                    for (DynamicDeploymentProcessorAdder adder : sl) {
+                        MicroProfileReactiveMessagingLogger.LOGGER.debugf("Invoking DynamicDeploymentProcessorAdder implementation: %s", adder);
+                        adder.addDeploymentProcessor(processorTarget, SUBSYSTEM_NAME);
+                    }
                 }
+
+
             }, RUNTIME);
 
             MicroProfileReactiveMessagingLogger.LOGGER.activatingSubsystem();
         }
+
+        private void installKafkaElytronSSLContextRegistryServiceIfPresent(OperationContext context) {
+            ClassLoader cl = WildFlySecurityManager.getClassLoaderPrivileged(this.getClass());
+            if (cl instanceof ModuleClassLoader) {
+                ModuleLoader loader = ((ModuleClassLoader)cl).getModule().getModuleLoader();
+                try {
+                    loader.loadModule("org.wildfly.reactive.messaging.kafka");
+                    ElytronSSLContextRegistryService.install(context.getServiceTarget(), context.getServiceRegistry(false));
```

Comment:
Without looking at the method being called, I assume this should be context.getServiceRegistry(true) as it will modify.

Reply:
No, this Service is just a wrapper around the service registry. It uses that to look up SSLContexts installed by the Elytron subsystem, so read-only is fine

Reply:
You've changed how this works, so my comment is no longer applicable.

But true was the correct value before -- it modified runtime state by installing a service, so the op needs the exclusive write lock, which 'true' triggers. It doesn't much matter as this OSH already gets the exclusive right lock, but I watch out for that so people don't see code and do the same in places where it might matter.

Reply:
For the record, it used `context.getServiceTarget()` to install the service itself. And (as it does now) it used the read-only registry to delegate the lookups to